synapse/rest/media/v1/media_repository.py

```diff
@@ -157,11 +159,31 @@ def _download_remote_file(self, server_name, media_id):
                 try:
                     length, headers = yield self.client.get_file(
                         server_name, request_path, output_stream=f,
-                        max_size=self.max_upload_size,
+                        max_size=self.max_upload_size, args={
+                            # tell the remote server to 404 if it doesn't
+                            # recognise the server_name, to make sure we don't
+                            # end up with a routing loop.
+                            "allow_remote": "false",
+                        }
                     )
+                except twisted.internet.error.DNSLookupError as e:
+                    logger.warn("HTTP error fetching remote media %s/%s: %r",
+                                server_name, media_id, e)
+                    raise NotFoundError()
+
+                except HttpResponseException as e:
+                    logger.warn("HTTP error fetching remote media %s/%s: %s",
+                                server_name, media_id, e.response)
+                    raise SynapseError.from_http_response_exception(e)
```

Comment:
I'm not convinced that we want to blindly forward the error from the other side without any sort of annotation. It makes sense to forward e.g. a 404, but what about a 400? A 401? In both cases it would probably be more appropriate to return a 500